headerfs/store.go

```diff
@@ -595,170 +438,91 @@ func (h *blockHeaderStore) ChainTip() (*wire.BlockHeader, uint32, error) {
 	return &latestHeader, tipHeight, nil
 }
 
-// FilterHeaderStore is an implementation of a fully fledged database for any
-// variant of filter headers.  The FilterHeaderStore combines a flat file to
-// store the block headers with a database instance for managing the index into
-// the set of flat files.
-type FilterHeaderStore struct {
-	*headerStore
-}
-
-// NewFilterHeaderStore returns a new instance of the FilterHeaderStore based
-// on a target file path, filter type, and target net parameters. These
-// parameters are required as if this is the initial start up of the
-// FilterHeaderStore, then the initial genesis filter header will need to be
-// inserted.
-func NewFilterHeaderStore(filePath string, db walletdb.DB,
-	filterType HeaderType, netParams *chaincfg.Params,
-	headerStateAssertion *FilterHeader) (*FilterHeaderStore, error) {
+// NewBlockHeaderStore creates a new instance of the blockHeaderStore based on
+// a target file path, an open database instance, and finally a set of
+// parameters for the target chain. These parameters are required as if this is
+// the initial start up of the blockHeaderStore, then the initial genesis
+// header will need to be inserted.
+func NewBlockHeaderStore(filePath string, db walletdb.DB,
+	netParams *chaincfg.Params) (BlockHeaderStore, error) {
 
-	fStore, err := newHeaderStore(db, filePath, filterType)
+	hStore, err := newHeaderStore(db, filePath, Block)
 	if err != nil {
 		return nil, err
 	}
 
-	// With the header store created, we'll fetch the fiie size to see if
+	// With the header store created, we'll fetch the file size to see if
 	// we need to initialize it with the first header or not.
-	fileInfo, err := fStore.file.Stat()
+	height, genesis, err := hStore.height()
 	if err != nil {
 		return nil, err
 	}
 
-	fhs := &FilterHeaderStore{
-		fStore,
+	bhs := &blockHeaderStore{
+		headerStore: hStore,
```

Comment:
Do you think we can make this headerStore field an interface? since we want to interact with the methods only and there are currently two implementations for this. Each implementation smartly separated by your use of build tags.